packages/element/src/Scene.ts

```diff
@@ -268,19 +267,13 @@ class Scene {
   }
 
   replaceAllElements(nextElements: ElementsMapOrArray) {
-    // ts doesn't like `Array.isArray` of `instanceof Map`
-    if (!isReadonlyArray(nextElements)) {
-      // need to order by fractional indices to get the correct order
-      nextElements = orderByFractionalIndex(
-        Array.from(nextElements.values()) as OrderedExcalidrawElement[],
-      );
-    }
-
+    // we do trust the insertion order on the map, though maybe we shouldn't and should prefer order defined by fractional indices
+    const _nextElements = toArray(nextElements);
```

Comment:
wait, we wanna keep the `!isReadonlyArray()` check because the common case is still to pass in arrays, and without the check we'd be copying it unnecessarily.

Reply:
perhaps we can change the `toArray()` to return the array as is if already array